host/xscope_io_host.c

```diff
@@ -75,16 +75,7 @@ void xscope_register(
 int send_file_chunk(unsigned file_idx, unsigned req_size)
 {
     unsigned char *buf = malloc(req_size);
-    unsigned n_bytes_read = 0;
-
-    n_bytes_read = fread(buf, 1, req_size, host_files[file_idx].fp);
-
-    if(n_bytes_read < req_size){
-        if(VERBOSE) printf("[HOST] Unexpected end of file, device requested: %u available: %u sent: 0\n", req_size, n_bytes_read);
-        xscope_ep_request_upload(END_MARKER_LEN, (const unsigned char *)end_sting); //End
-        free(buf);
-        return(-1);
-    }
+    unsigned n_bytes_read = fread(buf, 1, req_size, host_files[file_idx].fp);
```

Comment:
I'm a bit worried about taking this check out. However I see that we don't even check for an error in send_file_chunk.. and we just skip the transfer. So maybe this is OK.

Reply:
I think I do see an issue. If the fread resulted in an error, it would return 0 but `feof` would not return true, so the END_MARKER wouldn't be sent. I think I should add a check of [ferror](https://en.cppreference.com/w/c/io/ferror)

Reply:
Ultimately it should behave as close as possible to what happens with stdio.